### PR TITLE
Implement reductions from #24024

### PR DESCRIPTION
**Pull request by jbrockmendel**
min/max signatures+docstrings are changed from #24024, instead match those of DatetimeIndexOpsMixin (with appropriate docstring edits)

The min/max implementations are unchanged from 24024.  In an upcoming pass I'll see if we can pawn any more of it off to nanops.

Tests are verbatim from 24024.

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -1381,6 +1382,71 @@ def _ensure_localized(self, arg, ambiguous='raise', nonexistent='raise',
...
+        Return the maximum value of the Array or maximum along
+        an axis.
+
+        See Also
+        --------
+        numpy.ndarray.max
+        Index.max : Return the maximum value in an Index.
+        Series.max : Return the maximum value in a Series.
+        """
+        # TODO: skipna is broken with max.
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py]: this was just fixed?

File: pandas/tests/arrays/test_datetimes.py [hunk 2]
```diff
@@ -90,3 +90,41 @@ def test_setitem_clears_freq(self):
                                         tz='US/Central'))
         a[0] = pd.Timestamp("2000", tz="US/Central")
         assert a.freq is None
+
+
+class TestReductions(object):
+
+    @pytest.mark.parametrize("tz", [None, "US/Central"])
```
Comment (jreback) [inline pandas/tests/arrays/test_datetimes.py]: in test_reductions?

**Review by jreback — CHANGES_REQUESTED** [review]
maybe out th treats in the new pandas/tests/reductions

File: pandas/core/arrays/datetimelike.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: I thought so too, but no.  I expect it won't be too tough to fix nanops to apply the DTI fix to DTA, will do so in an upcoming pass after the 24024-specific parts are de-duplicated

File: pandas/tests/arrays/test_datetimes.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/test_datetimes.py]: before long, yah.  For now i want to make rebasing 24024 easy on tom

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ping on green.

**Comment by jbrockmendel** [comment]
this will need rebase following #24476.

**Comment by jbrockmendel** [comment]
Ping

File: pandas/core/arrays/datetimelike.py [hunk 3]
```diff
@@ -1381,6 +1382,71 @@ def _ensure_localized(self, arg, ambiguous='raise', nonexistent='raise',
...
+            return op(axis=axis, skipna=skipna, **kwargs)
+        else:
+            raise TypeError("cannot perform {name} with type {dtype}"
+                            .format(name=name, dtype=self.dtype))
+            # TODO: use super(DatetimeLikeArrayMixin, self)._reduce
+            #  after we subclass ExtensionArray
+
+    def min(self, axis=None, skipna=True, *args, **kwargs):
+        """
+        Return the minimum value of the Array or minimum along
```
Comment (jreback) [inline pandas/core/arrays/datetimelike.py]: are these meant in inherit docstrings? these should have Parameters if not

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/datetimelike.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: these are copied from the DatetimeIndexOps versions, with "Index" changes to "Array" and Index.min/Index.max added to the See Also sections.  These will be templated/shared before long hopefully

Comment (jreback) [reply pandas/core/arrays/datetimelike.py]: ok can u make a follow up issue for this

**Pull request merged**